test/tests/functional/pbs_checkpoint.py

```diff
@@ -0,0 +1,94 @@
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+
+
+class TestCheckpoint(TestFunctional):
+    """
+    This test suite targets Checkpoint functionality.
+    """
+
+    def num_compute_vnodes(self):
+        """
+        Count the Cray compute nodes and return the value.
+        """
+        vlist = []
+        vnl = self.server.filter(
+            VNODE, {'resources_available.vntype': 'cray_compute'})
+        vlist = vnl["resources_available.vntype=cray_compute"]
+        nv = len(vlist)
+        self.assertNotEqual(nv, 0, "There are no cray_compute vnodes present.")
+        return nv
+
+    def test_checkpoint_abort_with_preemept(self):
+        abort_script = """#!/bin/bash
+kill $1
+exit 0
+"""
+        abort_file = self.du.create_temp_file(body=abort_script)
+        self.du.chmod(path=abort_file, mode=0755)
+        c = {'$action': 'checkpoint_abort 30 !' + abort_file + ' %sid'}
+        self.mom.add_config(c)
+        self.scheduler.set_sched_config({'preempt_order': 'C'})
+        a = {'queue_type': 'e',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 200}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "expressq")
+        self.platform = DshUtils().get_platform()
+        if self.platform != 'cray' and self.platform != 'craysim':
+            node_attrs = self.server.status(NODE, id=self.server.shortname)[0]
+            ncpus = node_attrs['resources_available.ncpus']
+            attrs = {'Resource_List.select': '1:ncpus=' + ncpus}
+        else:
+            nv = self.num_compute_vnodes()
+            attrs = {ATTR_l + '.select': '%d:ncpus=1' % nv,
+                     ATTR_l + '.place': 'scatter'}
+
+        j1 = Job(TEST_USER, attrs)
+        j1.set_sleep_time(5)
```

Comment:
Depending on the speed of the system, isn't it possible j1 could be finished before j2 can cause it to be checkpointed.
Also, if the test is running on a Cray system, will this call affect the Cray default job's aprun sleep of 100 sec?

Reply:
makes sense.

Reply:
I will increase the sleep time to 100 seconds.

Reply:
I believe the default sleep time is 100s.  If you have no reason to change it, I'd leave it as the default(don't call set_sleep_time()).  The main reason to change the sleep time is you want to wait until the jobs end.  I don't think that is the case here.

Reply:
To be on the safer side, I have increased it to 1000s.